python/semantic_kernel/connectors/memory/azure_ai_search/utils.py

```diff
@@ -145,15 +145,21 @@ def data_model_definition_to_azure_ai_search_index(
                     algorithm_configuration_name=algo_name,
                 )
             )
-            algo_class, algo_params = INDEX_ALGORITHM_MAP[field.index_kind or "default"]
-            search_algos.append(
-                algo_class(
-                    name=algo_name,
-                    parameters=algo_params(
-                        metric=DISTANCE_FUNCTION_MAP[field.distance_function or "default"],
-                    ),
+            try:
+                algo_class, algo_params = INDEX_ALGORITHM_MAP[field.index_kind or "default"]
+            except KeyError as e:
+                raise ServiceInitializationError(f"Error: {e} not found in INDEX_ALGORITHM_MAP.") from e
+            try:
+                search_algos.append(
+                    algo_class(
+                        name=algo_name,
+                        parameters=algo_params(
+                            metric=DISTANCE_FUNCTION_MAP[field.distance_function or "default"],
+                        ),
+                    )
                 )
-            )
+            except KeyError as e:
+                raise ServiceInitializationError(f"Error: {e} not found in DISTANCE_FUNCTION_MAP.") from e
```

Comment:
nit: `e` is the exception not the key. 
nit: wrap the try block around the statement that accesses the map to make it more explicit.